### DISCUSSION TITLE
Responsive Layout in Flet

### DISCUSSION THREAD

**Discussion by ssinghm32** [flet-dev/flet#216] [General]
Just came across Flet from Hacker News and impressed with what I've seen so far.

Can Flet do responsive UI designs?  If so, how?

**Comment by FeodorFitsner** [comment]
Thanks for your interest!

Take a look at this example: https://github.com/flet-dev/examples/tree/main/python/community/responsive_menu_layout and let me know if it's something you are thinking of.
We also have plans to implement sort of "ResponsiveGrid" control similar to Bootstrap's responsive grid system.

Btw, in what HN post did you read about Flet? ;)

**Reply by ssinghm32** [reply]
Hi @FeodorFitsner ,

1) Thanks for the link to the Responsive Menu Layout.  That will definitely come in handy.
I was however looking for the "ResponsiveGrid" control, similar to Bootstrap :)  
Curious if the current Row and Column controls with wrapping will achieve something similar ??

2) Here is the link on HackerNews
https://news.ycombinator.com/item?id=32548900

**Reply by FeodorFitsner** [reply]
OK, then I'm going to implement `ResponsiveGrid` :)

**Reply by FeodorFitsner** [reply]
Subscribe to this issue: [ResponsiveGrid control](https://github.com/flet-dev/flet/issues/227)

**Reply by ssinghm32** [reply]
That's great @FeodorFitsner 

Curious why we can't use the GridView to achieve a responsive layout?

https://flet.dev/docs/guides/python/large-lists#gridview

**Reply by FeodorFitsner** [reply]
GridView is for lists where all cells will have the same size.

**Comment by pamoroso** [comment]
I don't know about Hacker News but Flet was featured in [issue 537 of PyCoder's Weekly](https://pycoders.com/issues/537).

**Reply by FeodorFitsner** [reply]
Oh, nice! Thanks for letting know!

**Comment by abcnow** [comment]
@FeodorFitsner .. I wonder what's going to be the strategy behind the new "responsive layout"? What I have seen in this link is very complicated and tedious https://github.com/flet-dev/examples/blob/main/python/community/responsive_menu_layout/responsive_menu_layout.py
so I am curious how are you going to tackle those challenges...

**Comment by abcnow** [comment]
PS: I think "fastui" has probably the most intuitive implementations when it comes to ui.. a few things there for inspiration...

https://fastui-demo.onrender.com/
